[EDU-1983] - Updates for new client options to handle domains and dns

GregHolmes opened the pull request:
## Description

surminus commented on content/root/platform-customization.textile:
```diff
@@ -40,7 +39,7 @@ As part of our Enterprise package offering, Ably are able to offer customers ded
 This involves the provision of dedicated realtime and routing servers, with shared database cluster and common services (such as stats, logging, registries, and configuration).
 
-We provide you with custom endpoints for your cluster, for example: @yourcompany-realtime.ably.io@ and @yourcompany-realtime.ably.io@.  Additionally, we can support CNAMEs using your own custom root domain.
+We provide you with custom endpoints for your cluster, for example: @yourcompany.realtime.ably.net@ and @yourcompany.rest.ably.net@.  Additionally, we can support CNAMEs using your own custom root domain.
```
The switch to the `endpoint` option means that we do not have separate domains for `realtime` and `rest` endpoints, they use the same domain.

```suggestion
We provide you with a custom endpoint for your cluster, for example: @yourcompany.realtime.ably.net@.  Additionally, we can support CNAMEs using your own custom root domain.
```

surminus commented on content/root/platform-customization.textile:
```diff
@@ -57,29 +56,31 @@ Note that this does not prevent customers outside of your chosen region from con
 
 h3. Decide whether to use an Ably subdomain, or your own domain
 
 Ably client libraries connect to two endpoints by default:
```
```suggestion
Ably client libraries connect to the following endpoint by default:
```

surminus commented on content/root/platform-customization.textile:
```diff
@@ -57,29 +56,31 @@ Note that this does not prevent customers outside of your chosen region from con
 
-- @realtime.ably.io@ - this endpoint is used for all WebSocket and other realtime protocol connections from the Ably client libraries.
-- @rest.ably.io@ - this endpoint is used for all REST-based requests and is optimized for all standard HTTP-based request types including all "REST requests":/docs/api/rest-api, "authentication requests":/docs/auth, and Comet and JSONP fallback when WebSockets cannot be used.
+- @realtime.ably.net@ - this endpoint is used for all WebSocket and other realtime protocol connections from the Ably client libraries.
+- @rest.ably.net@ - this endpoint is used for all REST-based requests and is optimized for all standard HTTP-based request types including all "REST requests":/docs/api/rest-api, "authentication requests":/docs/auth, and Comet and JSONP fallback when WebSockets cannot be used.
```
Probably want to rephrase this suggestion but basically we only have a single endpoint now, we do not have a separate one for REST.

```suggestion
- @main.realtime.ably.net@ - this endpoint is used by both WebSocket and other realtime protocol connections from the Ably client libraries and all REST-based requests and is optimized for all standard HTTP-based request types including all "REST requests":/docs/api/rest-api, "authentication requests":/docs/auth, and Comet and JSONP fallback when WebSockets cannot be used.
```

surminus commented on content/root/platform-customization.textile:
```diff
@@ -57,29 +56,31 @@ Note that this does not prevent customers outside of your chosen region from con
-These use Ably's existing wildcard certificate, which supports @*.ably.io@ and @*.ably-realtime.com@.
+For "example,":#modify @example.rest.ably.net@ and @example.realtime.ably.net@.
 
+These use Ably's existing wildcard certificate, which supports @*.ably.net@ and @*.ably-realtime.com@.
```
This isn't strictly true. We have wildcard certificates for `*.realtime.ably.net` and a variety of wildcards for different fallback subdomains, eg `*.a.fallback.main.cluster.ably-realtime.com` -> `*.a.fallback.main.cluster.ably-realtime.com`.

surminus commented on content/root/platform-customization.textile:
```diff
@@ -101,28 +102,26 @@ In addition, Ably handle all certificate renewals, rolling out to our global end
-* @fallbackHostsUseDefault@: whether to use the standard fallback hosts when an error requires the use of an alternative host. If you have a single host for both realtime and REST connections and specify that hostname using the @environment@ flag, then setting this option to @true@ will automatically determine which fallback hosts to use.
-* @fallbackHosts@ : an array of custom fallback hosts to try if an error requires the use of an alternative host. Ensure that you set these manually if specifying @realtimeHost@ and @restHost@ instead of @environment@.
+* a routing policy:
+  * @endpoint=main@ routes to `main.realtime.ably.net` (or `realtime.ably.net`)
```
We do not have an alias for `realtime.ably.net` (I think I commented in the ADR, but I guess that wasn't updated)

```suggestion
  * @endpoint=main@ routes to `main.realtime.ably.net`
```

GregHolmes commented:
Thank you @surminus, I've updated and added all changes in the following [fixup](https://github.com/ably/docs/pull/2651/commits/3554bf497ef022873e64f5de5e3813805f6ab9c1).

surminus commented:
LGTM. BTW I don't think this `endpoint` option has been released in ably-js yet, but I'm sure y'all aware :)

ttypic commented on content/root/platform-customization.textile:
```diff
@@ -57,29 +56,28 @@ Note that this does not prevent customers outside of your chosen region from con
 h3. Decide whether to use an Ably subdomain, or your own domain
 
-Ably client libraries connect to two endpoints by default:
+Ably client libraries connect to the @main.realtime.ably.net@ endpoint by default. This endpoint is used by both WebSocket and other realtime connections from the Ably client libraries, as well as all REST-based requests. It is optimized for all standard HTTP-based request types, including all "REST requests":/docs/api/rest-api, "authentication requests":/docs/auth, and Comet and JSONP fallback w ...
```
Once again endpoint used in several different meanings, I would use domain, host, etc.

ttypic reviewed (approved):
lgtm
